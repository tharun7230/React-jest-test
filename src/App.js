import _ from 'lodash';
import uuid from 'uuid';
import React from 'react';
import Text from './components/Text';
import TodoItem from './components/TodoItem';
import withSomething from './components/HOC/withSomething';

export class App extends React.Component {
  state = {
    todoItems: [
      { id: 1, title: "This is a todo!", isCompleted: true },
    ],
    inputValue: '',
  };

  handleTodoDelete = id => () =>
    this.setState({
      todoItems: _.filter(this.state.todoItems, todo => todo.id !== id),
    });

  handleTodoToggleComplete = id => () =>
    this.setState({
      todoItems: _.map(this.state.todoItems, todo => ({
        ...todo,
        isCompleted: id === todo.id ? !todo.isCompleted : todo.isCompleted,
      })),
    });

  createTodoItem = title => ({
    id: uuid.v4(),
    title,
    isCompleted: false,
  });

  handleAddTodo = (event) => {
    event.preventDefault();

    const { todoItems, inputValue } = this.state;

    this.setState({
      todoItems: [...todoItems, this.createTodoItem(inputValue)],
      inputValue: '',
    });
  }

  handleInputChange = ({ target }) =>
    this.setState({
      inputValue: target.value,
    });

  mapTodoItemsToTodoComponents = todoItems =>
    _.map(todoItems, todo => (
      <TodoItem
        id={todo.id}
        key={todo.id}
        title={todo.title}
        isCompleted={todo.isCompleted}
        onTodoDelete={this.handleTodoDelete}
        onToggleTodoComplete={this.handleTodoToggleComplete}
      />
    ));

  render() {
    const { todoItems, inputValue } = this.state;

    return (
      <div className="root anotherclass class">
        <div className="todo-container">
          <span className="todo-container-title">Todo Items</span>
          <div className="todo-input-box">
            <span>Create todo item</span>
            <Text
              onInputChange={this.handleInputChange}
              onAddTodo={this.handleAddTodo}
              value={inputValue}
            />
          </div>
          {this.mapTodoItemsToTodoComponents(todoItems)}
        </div>
      </div>
    );
  }
}

export default withSomething(App);
