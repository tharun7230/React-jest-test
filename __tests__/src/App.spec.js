import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App, { App as AppComponent } from '../../src/App';

describe('App', () => {
  it('handleInputChange should update state.inputValue', () => {
    const wrapper = shallow(<AppComponent />).instance();

    wrapper.handleInputChange({ target: { value: 'Hello YouTube' } });
    
    expect(wrapper.state.inputValue).toBe('Hello YouTube');
  });

  it('handleAddTodo should add a todo', () => {
    const wrapper = shallow(<AppComponent />).instance();
    const preventDefaultSpy = jest.fn();

    expect(wrapper.state.todoItems.length).toBe(1);

    wrapper.handleInputChange({ target: { value: 'Hello YouTube' } });
    expect(wrapper.state.inputValue).toBe('Hello YouTube');

    wrapper.handleAddTodo({ preventDefault: preventDefaultSpy });

    expect(wrapper.state.todoItems.length).toBe(2);
    expect(preventDefaultSpy.mock.calls.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<AppComponent />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
