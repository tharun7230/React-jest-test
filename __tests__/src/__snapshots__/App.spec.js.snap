// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="root anotherclass class"
>
  <div
    className="todo-container"
  >
    <span
      className="todo-container-title"
    >
      Todo Items
    </span>
    <div
      className="todo-input-box"
    >
      <span>
        Create todo item
      </span>
      <form
        onSubmit={[Function]}
      >
        <input
          className="input"
          onChange={[Function]}
          value=""
        />
      </form>
    </div>
    <div
      className="todo-item--completed"
    >
      <div
        className="todo-title"
      >
        This is a todo!
      </div>
      <div
        className="todo-actions"
      >
        <button
          className="todo-complete"
          onClick={[Function]}
        >
          Un-complete
        </button>
        <button
          className="todo-delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
